hilti/include/rt/types/vector.h

```diff
@@ -75,98 +79,256 @@ bool operator!=(Allocator<T, D1> const&, Allocator<U, D2> const&) noexcept {
     return false;
 }
 
-template<typename T, typename Allocator = std::allocator<T>>
-class SafeIterator
-    : public hilti::rt::detail::iterator::SafeIterator<
-          Vector<T, Allocator>, typename Vector<T, Allocator>::SafeIterator, SafeIterator<T, Allocator>> {
+template<typename T, typename Allocator>
+class Iterator {
+    using V = Vector<T, Allocator>;
+
+    std::weak_ptr<Vector<T, Allocator>*> _control;
+    typename V::size_type _index = 0;
+
+    using reference = T&;
```

Comment:
Let's move these definitions into the container where they belong. Ditto for `ConstIterator::reference`.